File: 330. Patching Array/greedy solution.py
class Solution(object):
    def minPatches(self, nums, n):
        """
        :type nums: List[int]
        :type n: int
        :rtype: int
        """
        miss, added, i = 1, 0, 0
        while miss <= n:
            if i < len(nums) and miss >= nums[i]:
                miss += nums[i]
                i += 1
            else:
                miss += miss
                added += 1
        return added

File: 330. Patching Array/greedy solution.java
public class Solution {
	public int minPatches(int[] nums, int n)
	{
		long miss = 1;
		int added = 0;
		int i = 0;

		while (miss <= n)
		{
			if ((i < nums.length) && (miss >= nums[i])) {
				miss += nums[i++];
			} else {
				miss += miss;
				added++;
			}
		}
		return (added);
	}
}
